lib/internal/modules/esm/fetch_module.js

```diff
@@ -148,7 +148,7 @@ function fetchWithRedirects(parsed) {
         err.message = `Cannot find module '${parsed.href}', HTTP 404`;
         throw err;
       }
-      if (res.statusCode < 200 || res.statusCode >= 400) {
+      if (res.statusCode < 200 || res.statusCode >= 300) {
         throw new ERR_NETWORK_IMPORT_DISALLOWED(
```

Comment:
Are you thinking correcting the error be handled separately/subsequently?

I think it would be better to create helpers for these, like Kidonng just added with `isRedirect()`. Something like

```js
if (!isStatusOk(res.statsuCode)) { // keeping with fetch's response.ok
  throw new ERR_NETWORK_IMPORT_BAD_RESPONSE(…);
```

I think `response.ok` is a bad prop name though because 200 = OK, and response.ok is any 2xx. If we're okay deviating, I think `isStatusGood` is a better name.

If you'd rather that also be a follow-up, sure, I can do.

A little troubling that no test needed to be updated for this 😨

Reply:
> I think `response.ok` is a bad prop name though because 200 = OK, and response.ok is any 2xx. If we're okay deviating, I think `isStatusGood` is a better name.

Why deviate? AFAICT web browsers would accept any response with a 2xx status code as valid (as long as it has the correct `Content-Type` header), I think Node.js should too.

Reply:
Do you mean what reason would we have for deviating, or why would using a different name be deviating?

I'm not proposing different status codes, merely word-smithing the name of the potential helper (`isStatusOk` vs `isStatusGood`).

Reply:
I'm sorry, I don't understand what you mean by "because 200 = OK, and response.ok is any 2xx", or how that's a justification for preferring `isStatusGood` – if anything that seems like a reason for keeping `ok` in the helper name.
Anyway, I don't have any strong feeling regarding the name of the helper, I interpreted your comment as you wanted to limit valid response to code `200`, as long as it's not the case I'm happy either way.

Reply:
> I interpreted your comment as you wanted to limit valid response to code 200

Noooo ;)

> I don't understand what you mean by "because 200 = OK, and response.ok is any 2xx"

Fetch's `response.ok` is `true` when the status code is 2xx. However, status code `200`'s textual name is `OK`. I'm wondering if people might misunderstand `isStatusOk` to mean "is the status specifically 200 OK". `isStatusGood` removes the ambiguity but deviates from fetch's poorly named `response.ok`.

Reply:
I'll be happy with any outcome that will be most appropriate according to "big picture" of this subsystem.

My personal preference for this part is to see it rewritten using `fetch` (i.e. undici). In that case, you can use `response.ok` directly and handle redirects without deviations from common ways. Additionally, you won't need to maintain this logic separately, e.g. if there will be new RFC with new codes.

As for naming and helper functions, my personal preference for `2xx` codes is to have it inlined with hardcoded numbers as long as there is only one place where it's checked; or relying on `response.ok` if possible.
Same for `3xx` if manual redirection handling will be kept: I think, having a function named `isRedirect()` or variable named `isRedirect` is exactly as self-explanatory in the code as needed.

Regarding `isStatus*` name, I agree that `response.ok` is poorly named. But if it's used on actual `Response`, it's ok. :)
`isStatusGood` sounds worse to me as it implies that there was some specific validation.
I'd suggest `isStatusSuccess` or `isStatusSuccessful` because it's canonical term for [`2xx` codes](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3).

Reply:
Oow, actually, that's an excellent point. Is there any reason we can't port this to actual fetch? Would we run into any cache contamination? 🤔 CORS would suddenly come into the picture.

Fetch didn't exist yet in Node when this was written, so that discussion hasn't already happened AFAIK.

Reply:
I think, porting this to `fetch` is feasible and desirable. CORS shouldn't be an issue for client-side Node.js.
Cache shouldn't be a problem as well: I don't know if ignoring cache-related HTTP headers was intentional, but [`Request.cache`](https://fetch.spec.whatwg.org/#concept-request-cache-mode) supports different modes, so you can either rely on it or handle manually.

Reply:
> I don't know if ignoring cache-related HTTP headers was intentional

I vaguely remember that it was but I couldn't tell you the rationale at the time.